heron/healthmgr/src/java/org/apache/heron/healthmgr/common/PhysicalPlanProvider.java

```diff
@@ -19,70 +19,129 @@
 
 package org.apache.heron.healthmgr.common;
 
+import java.net.HttpURLConnection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Collection;
 import java.util.logging.Logger;
 
 import javax.inject.Inject;
 import javax.inject.Named;
 import javax.inject.Provider;
 
-import com.microsoft.dhalion.events.EventHandler;
-import com.microsoft.dhalion.events.EventManager;
-
-import org.apache.heron.healthmgr.common.HealthManagerEvents.ContainerRestart;
-import org.apache.heron.healthmgr.common.HealthManagerEvents.TopologyUpdate;
+import org.apache.heron.api.generated.TopologyAPI;
 import org.apache.heron.proto.system.PhysicalPlans.PhysicalPlan;
+import org.apache.heron.proto.tmaster.TopologyMaster;
 import org.apache.heron.spi.statemgr.SchedulerStateManagerAdaptor;
+import org.apache.heron.spi.utils.NetworkUtils;
 
 import static org.apache.heron.healthmgr.HealthPolicyConfig.CONF_TOPOLOGY_NAME;
 
 /**
- * A topology's physical plan may get updated after initial deployment. This provider is used to
- * fetch the latest version from the state manager and provide to any dependent components.
+ * A topology's physical plan may get updated at runtime. This provider is used to
+ * fetch the latest version from the tmaster and provide to any dependent components.
  */
 public class PhysicalPlanProvider implements Provider<PhysicalPlan> {
   private static final Logger LOG = Logger.getLogger(PhysicalPlanProvider.class.getName());
 
   private final SchedulerStateManagerAdaptor stateManagerAdaptor;
   private final String topologyName;
 
-  private PhysicalPlan physicalPlan;
+  private PhysicalPlan cachedPhysicalPlan = null;
```

Comment:
the "Cached" feels a bit strange in this PR. It could be helpful to add a comment about when the data is cached and for how long.

Reply:
added comment